main/src/ca/uwaterloo/flix/api/lsp/provider/CompleteProvider.scala

```diff
@@ -54,8 +116,34 @@ object CompleteProvider {
     // Declaration-based:
 
     // Expressed-based:
-    CompletionItem("match", "match ${1:exp} {\n case ${2:pat} => ${3:exp}\n}", None, None, CompletionItemKind.Snippet, InsertTextFormat.Snippet, Nil),
-    CompletionItem("query", "query ${1:db} select ${2:cols} from ${3:preds} ${4:where ${5:cond}}", None, None, CompletionItemKind.Snippet, InsertTextFormat.Snippet, Nil),
+    CompletionItem("def", "def ${1:name}(${2:arg}:${3:type}): ${4:return_type} = \n", None, Some("code snippet to define a function"), CompletionItemKind.Snippet, InsertTextFormat.Snippet, Nil),
+    CompletionItem("elif", "else if (${1:/* condition */}) {\n    ${2:/* code */}\n}", None, Some("code snippet for else if ()"), CompletionItemKind.Snippet, InsertTextFormat.Snippet, Nil),
+    CompletionItem("else", "else {\n    ${1:/* code */}\n}", None, Some("code snippet for else"), CompletionItemKind.Snippet, InsertTextFormat.Snippet, Nil),
+    CompletionItem("enum", "enum ${1:name} {\n    case ${2:DataType1}\n    ${4:/* add more algebric data types separated by comma (,) */}\n}", None, Some("code snippet for enum"), CompletionItemKind.Snippet, InsertTextFormat.Snippet, Nil),
```

Comment:
Why does the count go from 1, 2 to 4? What about 3?

Reply:
I would rename `DataType1` to `name` and then I would drop the comment.

Reply:
this seems fine:
```
enum ${1:Name} {
	case ${2:name}
}
```
snippet for this:
`enum ${1:Name} {\n\tcase ${2:name}\n}`